ga4gh/datarepo.py

```diff
@@ -385,6 +388,11 @@ def _readSystemTable(self, cursor):
         row = cursor.fetchone()
         self._repositoryVersion = config["repositoryVersion"]
         self._creationTimeStamp = config["creationTimeStamp"]
+        if self._repositoryVersion != self.version:
```

Comment:
We may as well do semver checking here from the start. If minor != current minor, fine. If major != current major, throw exception.